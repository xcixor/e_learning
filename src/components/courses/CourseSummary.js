import React from "react";
import PropTypes from "prop-types";

const CourseSummary = ({ course, author }) => {
	return (
		<div className="jumbotron">
			<h2 className="display-6">{course.title}</h2>
			<h3 className="display-7">{author.name}</h3>
			<p className="lead">
				Course information.
			</p>
			<p className="lead">
				<a className="btn btn-primary btn-lg" href="#" role="button">
					Start Course
				</a>
			</p>
		</div>
	);
};

CourseSummary.propTypes = {
	course: PropTypes.object.isRequired,
	author: PropTypes.object.isRequired,
};

export default CourseSummary;
