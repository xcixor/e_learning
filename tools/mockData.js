const courses = [
	{
		id: 1,
		title: "Securing React Apps with Auth0",
		slug: "react-auth0-authentication-security",
		authorId: 1,
		category: "JavaScript",
	},
	{
		id: 2,
		title: "React: The Big Picture",
		slug: "react-big-picture",
		authorId: 1,
		category: "JavaScript",
	},
	{
		id: 3,
		title: "Creating Reusable React Components",
		slug: "react-creating-reusable-components",
		authorId: 1,
		category: "JavaScript",
	},
	{
		id: 4,
		title: "Building a JavaScript Development Environment",
		slug: "javascript-development-environment",
		authorId: 1,
		category: "JavaScript",
	},
	{
		id: 5,
		title: "Building Applications with React and Redux",
		slug: "react-redux-react-router-es6",
		authorId: 1,
		category: "JavaScript",
	},
	{
		id: 6,
		title: "Building Applications in React and Flux",
		slug: "react-flux-building-applications",
		authorId: 1,
		category: "JavaScript",
	},
	{
		id: 7,
		title: "Clean Code: Writing Code for Humans",
		slug: "writing-clean-code-humans",
		authorId: 1,
		category: "Software Practices",
	},
	{
		id: 8,
		title: "Architecting Applications for the Real World",
		slug: "architecting-applications-dotnet",
		authorId: 1,
		category: "Software Architecture",
	},
	{
		id: 9,
		title: "Becoming an Outlier: Reprogramming the Developer Mind",
		slug: "career-reboot-for-developer-mind",
		authorId: 1,
		category: "Career",
	},
	{
		id: 10,
		title: "Web Component Fundamentals",
		slug: "web-components-shadow-dom",
		authorId: 1,
		category: "HTML5",
	},
];

const chapters = [
	{
		id: 1,
		title: "Course Overview",
		slug: "course-overview",
		chapterNumber: 1,
		courseId: 1,
	},
	{
		id: 2,
		title: "Introduction",
		slug: "introduction",
		chapterNumber: 2,
		courseId: 1,
	},
	{
		id: 3,
		title: "Environment Setup",
		slug: "environment-setup",
		chapterNumber: 3,
		courseId: 1,
	},
	{
		id: 4,
		title: "Authorization and Authentication Standards",
		slug: "authorization-and-authentication-standards",
		chapterNumber: 4,
		courseId: 1,
	},
	{
		id: 5,
		title: "Login, Logout and Signup",
		slug: "login-logout-and-signup",
		chapterNumber: 5,
		courseId: 1,
	},
	{
		id: 1,
		title: "Course Overview",
		slug: "course-overview",
		chapterNumber: 1,
		courseId: 2,
	},
	{
		id: 2,
		title: "Introduction",
		slug: "introduction",
		chapterNumber: 2,
		courseId: 2,
	},
	{
		id: 3,
		title: "Environment Setup",
		slug: "environment-setup",
		chapterNumber: 3,
		courseId: 2,
	},
];

const authors = [
	{ id: 1, name: "Cory House" },
	{ id: 2, name: "Scott Allen" },
	{ id: 3, name: "Dan Wahlin" },
];

const newCourse = {
	id: null,
	title: "",
	authorId: null,
	category: "",
};

const newCourseChapter = {
	id: null,
	title: "",
	courseId: null,
	chapterNumber: null,
};

// Using CommonJS style export so we can consume via Node (without using Babel-node)
module.exports = {
	newCourse,
	courses,
	authors,
	chapters,
	newCourseChapter,
};
