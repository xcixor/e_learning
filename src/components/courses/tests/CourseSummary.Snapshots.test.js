import React from "react";
import CourseSummary from "../CourseSummary";
import renderer from "react-test-renderer";
import { courses, authors } from "../../../../tools/mockData";

const course = courses[0];
const author = authors[0];

it("renders", () => {
	const tree = renderer.create(
		<CourseSummary course={course} author={author} />
	);
	expect(tree).toMatchSnapshot();
});
