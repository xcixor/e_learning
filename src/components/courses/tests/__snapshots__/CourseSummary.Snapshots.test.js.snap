// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="jumbotron"
>
  <h2
    className="display-6"
  >
    Securing React Apps with Auth0
  </h2>
  <h3
    className="display-7"
  >
    Cory House
  </h3>
  <p
    className="lead"
  >
    Course information.
  </p>
  <p
    className="lead"
  >
    <a
      className="btn btn-primary btn-lg"
      href="#"
      role="button"
    >
      Start Course
    </a>
  </p>
</div>
`;
